src/NHibernate/Logging.cs

```diff
@@ -117,13 +117,8 @@ private static string GetNhibernateLoggerClass()
 			string nhibernateLoggerClass = null;
 			if (string.IsNullOrEmpty(nhibernateLogger))
 			{
-				// look for log4net.dll
-				string baseDir = AppDomain.CurrentDomain.BaseDirectory;
-				string relativeSearchPath = AppDomain.CurrentDomain.RelativeSearchPath;
-				string binPath = relativeSearchPath == null ? baseDir : Path.Combine(baseDir, relativeSearchPath);
-				string log4NetDllPath = binPath == null ? "log4net.dll" : Path.Combine(binPath, "log4net.dll");
-
-				if (File.Exists(log4NetDllPath) || AppDomain.CurrentDomain.GetAssemblies().Any(a => a.GetName().Name == "log4net"))
+				// look for log4net
+				if (System.Type.GetType("log4net.LogManager, log4net") != null)
 				{
 					nhibernateLoggerClass = typeof(Log4NetLoggerFactory).AssemblyQualifiedName;
 				}
```

Comment:
I am tempted to put a swallowing `try catch` there, but in case of failure we would have no way of knowing what happens excepted debugging. (No way to log anything in such a `catch` here, since that is the logging system which fails...)